### PR TITLE
Support configuring strict loading

### LINKED ISSUE #13246: CheckpointConnector should support loading with `strict=False`
## 🐛 Bug

pytorch_lightning/trainer/connectors/checkpoint_connector.py should support loading with `strict=False`

### Expected behavior

Say if you add a new torchmetrics module into your LightningModule, you wont be able to validate / test existing checkpoints because new keys are added.

This happens to all Trainer functions involving ckpt_path as parameter, such as `validate(model=None, dataloaders=None, ckpt_path=None, verbose=True, datamodule=None) `

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @borda @tchaton @justusschock @awaelchli @ananthsub @ninginthecloud @rohitgr7 @otaj

### LINKED ISSUE #14879: Allow `strict=False` in resuming `trainer.fit()` from a checkpoint
## 🚀 Feature

Add a `strict=False` option when using `trainer.fit(model,ckpt_path)`

### Motivation

I'm trying to pretrain a model and then load it to a different model (different state_dict and metrics), but when I do `trainer.fit(model.ckpt_path)` I get an error saying the state_dict is different. I know it is different and I want it to be, but I can't pass strict=False as I do with `model.load_from_checkpoint(path,strict=False).`

### Alternatives

Allowing somehting like: `trainer.fit(model,ckpt_path,strict=False)`

If there is already something that does this I couldn't find it.

cc @borda @justusschock @awaelchli @ninginthecloud @rohitgr7

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

WIP

Fixes #14879
Fixes #13246

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
stale pr policy
